woocommerce/woocommerce-ios
# Adding a package UI - Add package name text field

Part of #13551


## Description


- Added package name text field in "Custom" package tab when toggle "save this as a new package template" is ON.
- Added basic field validation, fields need to be non empty for buttons to be enabled.

Features remaining for the "Custom" tab:
- Saving package template logic
- Add package - gather all info from fields and using it for the package
- Use measure type that we got from backend (centimeters, inches etc.)

Features remaining for the "Carrier" tab:
- all

Features remaining for the "Saved" tab:
- all

Rest of the functionalities will be done in **separate** stacked PRs (done on top of this branch or trunk if this gets merged before).

## Testing information


1. Install and set up the Woo Shipping extension on your store.
2. Build and run the app with the `revampedShippingLabelCreation` feature flag enabled.
3. Create an order with the processing status and at least one physical product.
4. In the order details, select "Create Shipping Label."
5. Tap "Select a Package" button
6. Tap on the package type button to change the type
7. Tap in length, width and height fields to input values and check that the "Add Package" button is enabled when all 3 values are entered
8. Toggle "save this as a new package template" ON and "Save package template" button will show with text field for name above it.
9. Type some text in the text field
10. "Save package template" should be enabled when all dimension fields are non empty and text field for package name is non empty.

## Screenshots


https://github.com/user-attachments/assets/c7fc71f5-4b84-42b3-a4e2-aa290e947f89

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
